test/sequential/feeLock.API.feePayment.test.ts

```diff
@@ -0,0 +1,399 @@
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const tokenFees = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountAfter.reserved!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.reserved!
+    );
+
+  const userMgaFees = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountAfter.free!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.free!
+    )
+    .add(new BN(sellAssetsValue));
+
+  expect(tokenFees).bnEqual(new BN(0));
+  expect(userMgaFees).bnEqual(new BN(0));
+});
+
+test("gassless- GIVEN a feeLock configured WHEN a swap happens THEN fees are not charged but locked instead", async () => {
+  await addMgaToWhitelisted();
+
+  await testUser1.addMGATokens(sudo);
+  testUser1.addAsset(MGA_ASSET_ID);
+
+  const sellAssetsValue = thresholdValue.sub(new BN(5));
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  await testUser1.sellAssets(MGA_ASSET_ID, firstCurrency, sellAssetsValue);
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const tokenFees = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountAfter.reserved!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.reserved!
+    );
+
+  const userMgaFees = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountAfter.free!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.free!
+    )
+    .add(new BN(feeLockAmount))
+    .add(new BN(sellAssetsValue));
+
+  expect(tokenFees).bnEqual(new BN(feeLockAmount));
+  expect(userMgaFees).bnEqual(new BN(0));
+});
+
+test("gassless- GIVEN a correct config for gassless swaps WHEN the user runs unlock-fee THEN fees are charged", async () => {
+  await addMgaToWhitelisted();
+
+  await testUser1.addMGATokens(sudo);
+  testUser1.addAsset(MGA_ASSET_ID);
+
+  const sellAssetsValue = thresholdValue.sub(new BN(5));
+
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  await testUser1.sellAssets(MGA_ASSET_ID, firstCurrency, sellAssetsValue);
+  await waitForNBlocks(periodLength);
+  await unlockFee(testUser1);
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+
+  const tokenBlocked = testUser1.getAsset(MGA_ASSET_ID)?.amountAfter.reserved!;
+
+  const userMgaFees = testUser1
+    .getAsset(MGA_ASSET_ID)
+    ?.amountAfter.free!.sub(
+      testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.free!
+    )
+    .add(new BN(sellAssetsValue));
+
+  expect(tokenBlocked).bnEqual(new BN(0));
+  expect(userMgaFees).bnEqual(new BN(0));
+});
+
+test("gassless- High-value swaps are rejected from the txn pool if they would fail before the percentage fee is charged", async () => {
+  await addMgaToWhitelisted();
+
+  await testUser1.addMGATokens(sudo);
+  testUser1.addAsset(MGA_ASSET_ID);
+
+  await checkErrorSellAsset(
+    "1010: Invalid Transaction: The swap prevalidation has failed",
+    toBN("1", 26),
+    false
+  );
+});
+
+test("gassless- For low-value swaps, token reservation status and pallet storage are altered in accordance with the timeout mechanism", async () => {
```

Comment:
This test is about ensure that the storage is right. Could you please also validate the state of the remaining items from the `accountFeeLockData` storage ?.

Reply:
Done

Reply:
Done